File: CS/E5039/Reports/ProductsReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;

namespace E5039.Reports {
    public partial class ProductsReport : DevExpress.XtraReports.UI.XtraReport {
        public ProductsReport() {
            InitializeComponent();
        }

        private void ProductsReport_DataSourceDemanded(object sender, EventArgs e) {
            productsTableAdapter1.Fill(nWindDataSet1.Products, (int?) Parameters["CategoryID"].Value);
        }

    }
}

File: VB/E5039/Reports/ProductsReport.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI

Namespace E5039.Reports
	Partial Public Class ProductsReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub ProductsReport_DataSourceDemanded(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.DataSourceDemanded
			productsTableAdapter1.Fill(nWindDataSet1.Products, CType(Parameters("CategoryID").Value, Integer?))
		End Sub

	End Class
End Namespace
